src_c/geometry.c

```diff
@@ -8,6 +8,106 @@
 
 #define PYGAMEAPI_GEOMETRY_NUMSLOTS 21
 
+static PyObject *
+pg_raycast(PyObject *_null, PyObject *const *args, Py_ssize_t nargs)
+{
+    double start_x, start_y;
+    double end_x, end_y;
+    PyObject **farr;
+    Py_ssize_t col_length;
+    Py_ssize_t loop;
+    double angle;
+    double max_dist;
+
+    if (nargs == 3) {
+        if (!pg_TwoDoublesFromObj(args[0], &start_x, &start_y)) {
+            return RAISE(PyExc_TypeError, "the starting position requires a pair of floats");
+        }
+        if (!pg_TwoDoublesFromObj(args[1], &end_x, &end_y)) {
+            return RAISE(PyExc_TypeError, "the end position requires a pair of floats");
+        }
+    }
+    else if (nargs == 4) {
+        if (!PySequence_FAST_CHECK(args[3])) {
+            return RAISE(PyExc_TypeError, "argument of raycast() must be a sequence");
+        }
```

Comment:
This is not needed, you had a duplicated check here i just noticed.

Reply:
Whoops.